extensions/azurecore/package.json

```diff
@@ -128,6 +128,156 @@
+            "description": "%config.providerSettingsJson%",
+            "scope": "resource",
+            "default": [
+              {
+                "name": "Default",
+                "settings": {
+                  "metadata": {
+                    "displayName": "DefaultCloud",
+                    "id": "azure_defaultCloud",
+                    "endpoints": {
+                      "host": "https://login.microsoftonline.com/",
+                      "microsoftResource": "https://management.core.windows.net/",
+                      "graphResource": "https://graph.windows.net/",
+                      "msGraphResource": "https://graph.microsoft.com/",
+                      "armResource": "https://management.azure.com/",
+                      "sqlResource": "https://database.windows.net/",
+                      "azureKeyVaultResource": "https://vault.azure.net/",
+                      "azureLogAnalyticsResource": "https://api.loganalytics.io/",
+                      "azureStorageResource": {
+                        "endpoint": "",
+                        "endpointSuffix": ".core.windows.net/"
+                      },
+                      "azureKustoResource": "https://kusto.kusto.windows.net/",
+                      "powerBiResource": "https://analysis.windows.net/powerbi/api/",
+                      "scopes": "https://management.azure.com/user_impersonation",
+                      "portalEndpoint": "https://portal.azure.com"
+                    }
+                  }
+                }
+              }
+            ],
+            "items": {
+              "type": "object",
+              "properties":{
+                "required": ["name", "settings"],
+                "name": {
+                  "type": "string",
+                  "default": "Azure Public",
+                  "description": "Cloud Name"
+                },
+                "settings": {
+                  "type": "object",
+                  "description": "Cloud Settings",
+                  "required": ["metadata"],
+                  "properties": {
+                    "metadata": {
+                      "type": "object",
+                      "required": ["displayName", "id"],
+                      "properties": {
+                        "displayName": {
+                          "type": "string",
+                          "default": "Azure Public Cloud",
+                          "description": "Cloud Display Name"
+                        },
+                        "id": {
+                          "type": "string",
+                          "default": "azure_publicCloud",
+                          "description": "Cloud ID"
+                        },
+                        "endpoints": {
+                          "type": "object",
+                          "required": ["host","scopes", "sqlResource", "microsoftResource", "armResource", "graphResource", "azureStorageResource", "azureKeyVaultResource"],
```

Comment:
Why is AKV, Storage resource required?

Reply:
This is because they were required by default in azurecore.d.ts and they are needed by some of the ADS extensions (for example, `getBlobs` requires storage account in the `getStorageAccountAccessKey` function)

Reply:
I don't think AKV is used anywhere - can we add a null check and add it to resource list like we do for others?